com.unity.multiplayer.mlapi/Runtime/Interest/InterestNode.cs

```diff
@@ -0,0 +1,104 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using MLAPI.Connection;
+using UnityEngine;
+using UnityEngine.Serialization;
+
+namespace MLAPI.Interest
+{
+    public abstract class InterestObjectStorage : ScriptableObject
+    {
+        public abstract void Query(in NetworkClient client, HashSet<NetworkObject> results);
+        public abstract void AddObject(NetworkObject obj);
+        public abstract void RemoveObject(NetworkObject obj);
+        public abstract void UpdateObject(NetworkObject obj);
+    };
+
+    // maybe seal this class
+    [CreateAssetMenu(fileName = "InterestNode", menuName = "Interest/Nodes/InterestNode", order = 1)]
+    [Serializable]
+    public class InterestNode : ScriptableObject
+    {
+        // set this delegate if you want a function called when
+        //  object 'obj' is being spawned / de-spawned
+        public delegate void SpawnDelegate(in NetworkObject obj);
+
+        public SpawnDelegate OnSpawn;
+        public SpawnDelegate OnDespawn;
+
+        public InterestSettings InterestSettings;
+        public InterestObjectStorage InterestObjectStorage;
+
+        public void AddObject(NetworkObject obj)
+        {
+            InterestObjectStorage?.AddObject(obj);
+        }
+        public void RemoveObject(NetworkObject obj)
+        {
+            InterestObjectStorage?.RemoveObject(obj);
+        }
+        public void UpdateObject(NetworkObject obj)
+        {
+            InterestObjectStorage?.UpdateObject(obj);
+        }
+
+        public InterestNode()
+        {
+            ChildNodes = new List<InterestNode>();
+        }
+
+        // externally-called object query function.
+        //  The passed-in hash set will contain the results.
+        public void QueryFor(in NetworkClient client, HashSet<NetworkObject> results)
+        {
+            InterestObjectStorage?.Query(client, results);
+
+            foreach (var c in ChildNodes)
+            {
+                c.QueryFor(client, results);
+            }
+        }
```

Comment:
this method body looks very similar to `InterestManager`s, isn't it?

Reply:
Yup, see above